da-indexer/da-indexer-logic/src/celestia/l2_router/arbitrum.rs

```diff
@@ -0,0 +1,66 @@
+use super::types::{L2BatchMetadata, L2Config};
+use anyhow::Result;
+use blockscout_display_bytes::Bytes;
+use chrono::DateTime;
+use reqwest::{Client, StatusCode, Url};
+use serde::Deserialize;
+
+#[derive(Deserialize, Debug)]
+struct CommitmentTransaction {
+    hash: String,
+    timestamp: String,
+}
+
+#[derive(Deserialize, Debug)]
+struct L2BatchArbitrum {
+    commitment_transaction: CommitmentTransaction,
+    end_block: u64,
+    number: u64,
+    start_block: u64,
+    transactions_count: u64,
+}
+
+pub async fn get_l2_batch(
+    config: &L2Config,
+    height: u64,
+    commitment: &[u8],
+) -> Result<Option<L2BatchMetadata>> {
+    let commitment = Bytes::from(commitment.to_vec()).to_string();
+    let query = format!(
+        "{}/api/v2/arbitrum/batches/da/celestia/{}/{}",
+        config.l2_api_url, height, commitment,
+    );
+
+    let response = Client::new()
+        .get(&query)
+        .timeout(config.request_timeout)
+        .send()
+        .await?;
```

Comment:
Usually we use `reqwest_retry` (https://docs.rs/reqwest-retry/latest/reqwest_retry/) client implementation when sending requests to 3-rd party services to handle some transient errors with retry pattern. Maybe you want to consider that here

E.g., https://github.com/blockscout/blockscout-rs/blob/main/sig-provider/sig-provider/src/sources/mod.rs#L58

Reply:
Yes, it makes sense, thank you! Fixed in https://github.com/blockscout/blockscout-rs/pull/999/commits/c88d14a797fd6f05354095167cfe55b6814ca6ca